### PR TITLE
[prometheus]  Update Mimir to v2.14.2 

### PR DESCRIPTION
**Pull request by ykav-flant**
## Description

Update Mimir component to v2.14.2 + update net/crypto packages to mitigate CVEs.

## Why do we need it, and what problem does it solve?
Improves security.

## What is the expected result?
Fixed CVEs in crypto and net packages.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: prometheus
type: chore
summary: Update Mimir net/crypto packages
impact_level: low
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @ykav-flant. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @ykav-flant. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @ykav-flant. I've started the workflow for label 'e2e/run/yandex-cloud'...

File: modules/300-prometheus/images/mimir/werf.inc.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    add: /src
+    before: install
+mount:
+  - fromPath: ~/go-pkg-cache
+    to: /go/pkg
+shell:
+  install:
+    - cd /src
+    - export GOPROXY={{ $.GOPROXY }} CGO_ENABLED=0 GOOS=linux GOARCH=amd64
+    - go mod tidy
```
Comment (RomanenkoDenys) [inline modules/300-prometheus/images/mimir/werf.inc.yaml:34]:
```suggestion
    - go mod download -x
```

**Review by RomanenkoDenys — APPROVED** [review]
sme nits

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @ykav-flant. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Review by RomanenkoDenys — APPROVED** [review]

**Review by vladimirGuryanov — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [11465](https://github.com/deckhouse/deckhouse/pull/11465) to the branch [release-1.67](https://github.com/deckhouse/deckhouse/tree/release-1.67) successful!

**Pull request merged**